### DISCUSSION TITLE
GSoC - NNCF Quantization Analytics Tool

### DISCUSSION THREAD

**Discussion by DaniAffCH** [openvinotoolkit/openvino#23177] [Google Summer of Code]
Hi, I'm Daniele, a student interested in contributing to openvino. 
I plan to apply to GSoC24  and I've been particularly attracted by the proposed project  "NNCF Quantization Analytics Tool". 
As per the [Prerequisite Task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task) section, I am required to fulfill a GFI. Could you please suggest me if there is a specific GFI task available that is directly related to the project?
Thank you in advance @alexsu52  @andrey-churkin

**Comment by Om-Doiphode** [comment]
Hi @alexsu52, @andrey-churkin, I am Om Doiphode, a third-year undergraduate in Computer Engineering at Veermata Jijabai Technological Institute (VJTI), Mumbai, India. I would like to work on the project `NNCF Quantization Analytics Tool`. Currently, I am an ML intern at an AI startup in Sunnyvale, California. Can you please guide me further on this project? Thanks!

**Reply by alexsu52** [reply]
Hi @Om-Doiphode! You should start with [Prerequisite Task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task). Unfortunately, all NNCF GFI issues already assigned, please, stay tune or take any another task form the [board](https://github.com/orgs/openvinotoolkit/projects/3/views/1).

**Comment by alexsu52** [comment]
> Hi, I'm Daniele, a student interested in contributing to openvino. I plan to apply to GSoC24 and I've been particularly attracted by the proposed project "NNCF Quantization Analytics Tool". As per the [Prerequisite Task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task) section, I am required to fulfill a GFI. Could you please suggest me if there is a specific GFI task available that is directly related to the project? Thank you in advance @alexsu52 @andrey-churkin

Hi @DaniAffCH! Thank you for your interest in the project. @andrey-churkin, could you submit GFI for @DaniAffCH?

**Reply by andrey-churkin** [reply]
@DaniAffCH Please take the following issue https://github.com/openvinotoolkit/nncf/issues/2552

**Reply by DaniAffCH** [reply]
Thank you! @andrey-churkin

**Comment by Viditagarwal7479** [comment]
Hi @alexsu52 a d @andrey-churkin I am also interested in "NNCF Quantization Analytics Tool" and was about to submit a same discussion thread as this, but thought to go through the ones already submitted and found this (really surprised). Can you please suggest me also a specific GFI task that is related to the project?

**Reply by alexsu52** [reply]
Hi @Viditagarwal7479! Thank you for your interest in the project. As I see, one GFI https://github.com/openvinotoolkit/nncf/issues/2564 is not assign right now, you can take. I'm sure that this is good [Prerequisite Task](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#prerequisite-task) for the submission.

**Reply by Viditagarwal7479** [reply]
Thank you @alexsu52 ^^

**Reply by Viditagarwal7479** [reply]
Hey @alexsu52, I have created my PR corresponding to the above GFI. In the meantime, it gets reviewed, would you like to suggest some tasks which might increase my chances for this project?

**Reply by zhuo-yoyowz** [reply]
@alexsu52  Please help review the PR and provide further suggestions. Thanks!

**Reply by andrey-churkin** [reply]
@Viditagarwal7479 Hi, you already have an assigned reviewer. It would be best to concentrate on the comments provided by this reviewer. Also, please keep track and feel free to take GFI that you would like more. I think it helps you get more familiar with NNCF. Please let me know if you have questions. Thanks!

**Comment by alexsu52** [comment]
> Hey @alexsu52, I have created my PR corresponding to the above GFI. In the meantime, it gets reviewed, would you like to suggest some tasks which might increase my chances for this project?

I would recommend taking a look at the quantization algorithm from NNCF because we will be using it in the GSoC task. This will also help you create more better proposal.

**Comment by alexsu52** [comment]
> @alexsu52 Please help review the PR and provide further suggestions. Thanks!

Could you share the link to the PR?
